extensions/bttv-emote/src/index.tsx

```diff
@@ -0,0 +1,172 @@
+        ))}
+      </List.Section>
+    </List>
+  );
+}
+
+function SearchListItem({ searchResult }: { searchResult: Emote }) {
+  const listImage = imageBaseURL + searchResult.id + "/1x.png";
+  const image1x = imageBaseURL + searchResult.id + "/1x." + searchResult.imageType;
+  const image2x = imageBaseURL + searchResult.id + "/2x." + searchResult.imageType;
+  const image3x = imageBaseURL + searchResult.id + "/3x." + searchResult.imageType;
+
+  const browserUrl = browserBaseURL + searchResult.id;
+
+  return (
+    <List.Item
+      title={searchResult.code}
+      icon={listImage}
+      subtitle={searchResult.imageType}
+      actions={
+        <ActionPanel>
+          <ActionPanel.Section>
+          <CopyToClipboardAction
+              title="Copy emote"
+              content={image2x}
+            />
+            <PasteAction
+              title="Paste emote"
+              content={image2x}
+            />
+          </ActionPanel.Section>
+          <ActionPanel.Section>
+            <CopyToClipboardAction
+              title="Copy 1x Emote"
+              content={image1x}
+              shortcut={{ modifiers: ["cmd"], key: "s" }}
+
+            />
+            <CopyToClipboardAction
+              title="Copy 2x Emote"
+              content={image2x}
+              shortcut={{ modifiers: ["cmd"], key: "m" }}
+
+            />
+            <CopyToClipboardAction
+              title="Copy 3x Emote"
+              content={image3x}
+              shortcut={{ modifiers: ["cmd"], key: "l" }}
+
+            />
+          </ActionPanel.Section>
+          <ActionPanel.Section>
+            <OpenInBrowserAction
+              title="Open in Browser"
+              url={browserUrl}
+              shortcut={{ modifiers: ["cmd"], key: "o" }}
+            />
+          </ActionPanel.Section>
+        </ActionPanel>
+      }
+    />
+  );
+}
+
+function useSearch() {
+  const [state, setState] = useState<SearchState>({ results: [], isLoading: true });
+  const cancelRef = useRef<AbortController | null>(null);
+
+  useEffect(() => {
+    search("");
+    return () => {
+      cancelRef.current?.abort();
+    };
+  }, []);
+
+  async function search(searchText: string) {
+    cancelRef.current?.abort();
+    cancelRef.current = new AbortController();
+    try {
+      setState((oldState) => ({
+        ...oldState,
+        isLoading: true,
+      }));
+      const results = await performSearch(searchText, cancelRef.current.signal);
+      setState((oldState) => ({
+        ...oldState,
+        results: results,
+        isLoading: false,
+      }));
+    } catch (error) {
+      if (error instanceof AbortError) {
+        return;
+      }
+      showToast(
+        ToastStyle.Failure, "Error", "Emote not found"
+      )
+    }
+  }
+
+  return {
+    state: state,
+    search: search,
+  };
+}
+
+async function performSearch(searchText: string, signal: AbortSignal): Promise<Emote[]> {
```

Comment:
When searching for less than three characters, it still makes a fetch call, that's unnecessary. It should not make any calls when it's not possible.